### ISSUE TITLE
0.9.1 hwdec=vdpau not works

### ISSUE THREAD

**Issue by spvkgn** [mpv-player/mpv#1948]
With 0.8.2 hwdec=vdpau works without any issues. Next versions 0.8.3 and 0.9.x does not enable hardware decoding with vdpau. Can you please help with this issue?

```
$ ./mpv -V
mpv 0.9.1 (C) 2000-2015 mpv/MPlayer/mplayer2 projects
 built on 2015-05-04T11:27:54
ffmpeg library versions:
   libavutil       54.7.100
   libavcodec      56.1.100
   libavformat     56.4.101
   libswscale      3.0.100
   libavfilter     5.1.100
   libswresample   1.1.100
```
```
$ ./mpv --no-config --vo=vdpau --hwdec=vdpau /media/MMEDIA/DVD/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ BDRip\ 720p\]/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ Progressive\ Metal,\ BDRip\ 720p\].mkv 
Playing: /media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p].mkv
 (+) Video --vid=1 (h264)
 (+) Audio --aid=1 --alang=eng (*) 'PCM 2.0 1536 Kbps 48.0 KHz 16 bits' (flac)
File tags:
 Title: Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p]
Using software decoding.
AO: [pulse] 48000Hz stereo 2ch s16
VO: [vdpau] 1280x720 yuv420p
AV: 00:00:04 / 02:56:42 (0%) A-V:  0.000 Cache: 10s+144635KB


Exiting... (Quit)
```
```
$ vdpauinfo 
display: :0   screen: 0
API version: 1
Information string: NVIDIA VDPAU Driver Shared Library  331.113  Mon Dec  1 20:30:22 PST 2014

Video surface:

name   width height types
-------------------------------------------
420     4096  4096  NV12 YV12 
422     4096  4096  UYVY YUYV 

Decoder capabilities:

name               level macbs width height
-------------------------------------------
MPEG1                 0 65536  4032  4048
MPEG2_SIMPLE          3 65536  4032  4048
MPEG2_MAIN            3 65536  4032  4048
H264_MAIN            41 65536  4032  4080
H264_HIGH            41 65536  4032  4080
VC1_SIMPLE            1  8190  2048  2048
VC1_MAIN              2  8190  2048  2048
VC1_ADVANCED          4  8190  2048  2048
MPEG4_PART2_SP        3  8192  2048  2048
MPEG4_PART2_ASP       5  8192  2048  2048
DIVX4_QMOBILE         0  8192  2048  2048
DIVX4_MOBILE          0  8192  2048  2048
DIVX4_HOME_THEATER    0  8192  2048  2048
DIVX4_HD_1080P        0  8192  2048  2048
DIVX5_QMOBILE         0  8192  2048  2048
DIVX5_MOBILE          0  8192  2048  2048
DIVX5_HOME_THEATER    0  8192  2048  2048
DIVX5_HD_1080P        0  8192  2048  2048

Output surface:

name              width height nat types
----------------------------------------------------
B8G8R8A8         16384 16384    y  Y8U8V8A8 V8U8Y8A8 
R10G10B10A2      16384 16384    y  Y8U8V8A8 V8U8Y8A8 

Bitmap surface:

name              width height
------------------------------
B8G8R8A8         16384 16384
R8G8B8A8         16384 16384
R10G10B10A2      16384 16384
B10G10R10A2      16384 16384
A8               16384 16384

Video mixer:

feature name                    sup
------------------------------------
DEINTERLACE_TEMPORAL             y
DEINTERLACE_TEMPORAL_SPATIAL     y
INVERSE_TELECINE                 y
NOISE_REDUCTION                  y
SHARPNESS                        y
LUMA_KEY                         y
HIGH QUALITY SCALING - L1        y
HIGH QUALITY SCALING - L2        -
HIGH QUALITY SCALING - L3        -
HIGH QUALITY SCALING - L4        -
HIGH QUALITY SCALING - L5        -
HIGH QUALITY SCALING - L6        -
HIGH QUALITY SCALING - L7        -
HIGH QUALITY SCALING - L8        -
HIGH QUALITY SCALING - L9        -

parameter name                  sup      min      max
-----------------------------------------------------
VIDEO_SURFACE_WIDTH              y         1     4096
VIDEO_SURFACE_HEIGHT             y         1     4096
CHROMA_TYPE                      y  
LAYERS                           y         0        4

attribute name                  sup      min      max
-----------------------------------------------------
BACKGROUND_COLOR                 y  
CSC_MATRIX                       y  
NOISE_REDUCTION_LEVEL            y      0.00     1.00
SHARPNESS_LEVEL                  y     -1.00     1.00
LUMA_KEY_MIN_LUMA                y  
LUMA_KEY_MAX_LUMA                y  
```

**Comment by wm4** [comment]
Post a log with -v, and also mediainfo output on the file if possible.

**Comment by spvkgn** [comment]
I have noticed it says "Requested hardware decoder not compiled" and I don't fully understand why this not compiled in 0.9.1 - I use the same build options as with 0.8.2.
```
$ ./mpv -v --no-config --vo=vdpau --hwdec=vdpau /media/MMEDIA/DVD/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ BDRip\ 720p\]/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ Progressive\ Metal,\ BDRip\ 720p\].mkv 
[cplayer] Command line options: '-v' '--no-config' '--vo=vdpau' '--hwdec=vdpau' '/media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p].mkv'
[cplayer] mpv 0.9.1 (C) 2000-2015 mpv/MPlayer/mplayer2 projects
[cplayer]  built on 2015-05-04T11:27:54
[cplayer] ffmpeg library versions:
[cplayer]    libavutil       54.7.100
[cplayer]    libavcodec      56.1.100
[cplayer]    libavformat     56.4.101
[cplayer]    libswscale      3.0.100
[cplayer]    libavfilter     5.1.100
[cplayer]    libswresample   1.1.100
[cplayer] 
[cplayer] Configuration: ./waf configure --enable-pdf-build
[cplayer] List of enabled features: alsa asm atomics audio-input avframe-metadata build-date cplayer debug-build dlopen drm dvbin dvdnav dvdread enca encoding fchmod gl gl-x11 glibc-thread-name glob iconv jpeg libass libass-osd libav libavdevice libavfilter libbluray libdl libm librt libswresample libv4l2 linux-fstatfs lua nanosleep optimize oss-audio oss-audio-native posix posix-or-mingw posix-spawn pthreads pulse pvr resampler rubberband shm stdatomic subprocess termios tv tv-v4l2 vaapi vaapi-glx vaapi-hwaccel vdpau vdpau-gl-x11 videodev vt.h x11 xext xinerama xrandr xss zlib
[cplayer] Setting option 'v' = '' (flags = 8)
[cplayer] Setting option 'no-config' = '' (flags = 8)
[cplayer] Setting option 'vo' = 'vdpau' (flags = 8)
[cplayer] Setting option 'hwdec' = 'vdpau' (flags = 8)
[input] Falling back on default (hardcoded) input config
[osc] Loading script @osc.lua...
[global] config file: 'scripts'
[global] config file: 'lua'
[osc] loading mp.defaults
[osc] loading @osc.lua
[global] config path: 'lua-settings/osc.conf' -> '(NULL)'
[osc] lua-settings/osc.conf not found. 
[cplayer] Done loading @osc.lua.
[ytdl_hook] Loading script @ytdl_hook.lua...
[global] config file: 'scripts'
[global] config file: 'lua'
[ytdl_hook] loading mp.defaults
[ytdl_hook] loading @ytdl_hook.lua
[cplayer] Run command: hook_add, flags=0, args=[on_load, 1, 10]
[cplayer] Done loading @ytdl_hook.lua.
[global] config file: 'scripts'
[global] config file: 'lua'
[global] config path: 'watch_later/25F5970BAD291FCA3B39DD80B324C4FA' -> '(NULL)'
[global] config path: 'watch_later' -> '(NULL)'
[global] config path: 'watch_later/25F5970BAD291FCA3B39DD80B324C4FA' -> '(NULL)'
[global] config path: 'watch_later' -> '(NULL)'
[cplayer] Playing: /media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p].mkv
[cplayer] Running hook: ytdl_hook/on_load
[cplayer] Run command: hook_ack, flags=0, args=[on_load]
[file] Opened: /media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p].mkv
[cache] Cache size set to 150000 KiB
[demux] Trying demuxer: disc (force-level: normal)
[demux] Trying demuxer: edl (force-level: normal)
[demux] Trying demuxer: cue (force-level: normal)
[demux] Trying demuxer: rawaudio (force-level: normal)
[demux] Trying demuxer: rawvideo (force-level: normal)
[demux] Trying demuxer: tv (force-level: normal)
[demux] Trying demuxer: libass (force-level: normal)
[demux] Trying demuxer: mkv (force-level: normal)
[mkv] Found the head...
[mkv] + a segment...
[mkv] /---- [ parsing seek head ] ---------
[mkv] \---- [ parsing seek head ] ---------
[mkv] |+ segment information...
[mkv] | + timecode scale: 1000000
[mkv] | + duration: 10602.816s
[mkv] | + segment uid 41 0a e1 af d4 40 63 1e 9d 2c 2b 6b 60 4f 17 ae
[mkv] |+ segment tracks...
[mkv] | + a track...
[mkv] |  + Track number: 1
[mkv] |  + Track type: Video
[mkv] |  + Video track
[mkv] |   + Display width: 1280
[mkv] |   + Display height: 720
[mkv] |   + Pixel width: 1280
[mkv] |   + Pixel height: 720
[mkv] |  + Codec ID: V_MPEG4/ISO/AVC
[mkv] |  + CodecPrivate, length 44
[mkv] |  + Default duration: 33.367ms ( = 29.970 fps)
[mkv] | + a track...
[mkv] |  + Track number: 2
[mkv] |  + Name: PCM 2.0 1536 Kbps 48.0 KHz 16 bits
[mkv] |  + Track type: Audio
[mkv] |  + Audio track
[mkv] |   + Sampling frequency: 48000.000000
[mkv] |   + Bit depth: 16
[mkv] |   + Channels: 2
[mkv] |  + Codec ID: A_FLAC
[mkv] |  + CodecPrivate, length 84728
[mkv] |  + Default duration: 85.333ms ( = 11.719 fps)
[mkv] /---- [ parsing chapters ] ---------
[mkv] New edition 0
[mkv] Default edition flag: 0
[mkv] Chapter 0 from 00:00:00.000 to 00:00:00.000, Intro
[mkv] Chapter 1 from 00:00:54.54000000 to 00:00:00.000, As I Am
[mkv] Chapter 2 from 00:08:35.715000000 to 00:00:00.000, This Dying Soul
[mkv] Chapter 3 from 00:20:47.512000000 to 00:00:00.000, Beyond This Life
[mkv] Chapter 4 from 00:40:22.520000000 to 00:00:00.000, Hollow Years
[mkv] Chapter 5 from 00:49:39.910000000 to 00:00:00.000, War Inside My Head
[mkv] Chapter 6 from 00:52:04.554000000 to 00:00:00.000, The Test That Stumped Them All
[mkv] Chapter 7 from 00:57:05.922000000 to 00:00:00.000, Endless Sacrifice
[mkv] Chapter 8 from 01:08:28.971000000 to 00:00:00.000, Instrumedley
[mkv] Chapter 9 from 01:20:36.398000000 to 00:00:00.000, Trial of Tears
[mkv] Chapter 10 from 01:34:34.835000000 to 00:00:00.000, New Millennium
[mkv] Chapter 11 from 01:42:35.616000000 to 00:00:00.000, Keyboard Solo
[mkv] Chapter 12 from 01:46:33.854000000 to 00:00:00.000, Only a Matter of Time
[mkv] Chapter 13 from 01:53:55.728000000 to 00:00:00.000, Goodnight Kiss
[mkv] Chapter 14 from 02:00:05.398000000 to 00:00:00.000, Solitary Shell
[mkv] Chapter 15 from 02:06:04.757000000 to 00:00:00.000, Stream of Consciousness
[mkv] Chapter 16 from 02:16:59.778000000 to 00:00:00.000, Disappear
[mkv] Chapter 17 from 02:22:54.532000000 to 00:00:00.000, Pull Me Under
[mkv] Chapter 18 from 02:32:23.601000000 to 00:00:00.000, In the Name of God
[mkv] \---- [ parsing chapters ] ---------
[mkv] |+ found cluster
[mkv] Deferring reading cues.
[mkv] All headers are parsed!
[mkv] Aspect: 1.777778
[demux] Detected file format: Matroska
[cache] blocking for STREAM_CTRL 8
[cache] blocking for STREAM_CTRL 8
[find_files] Loading external files in /media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/
[global] config path: 'sub/' -> '(NULL)'
[cplayer]  (+) Video --vid=1 (h264)
[cplayer]  (+) Audio --aid=1 --alang=eng (*) 'PCM 2.0 1536 Kbps 48.0 KHz 16 bits' (flac)
[cplayer] File tags:
[cplayer]  Title: Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p]
[cplayer] [V] fourcc:0x31637661  size:1280x720  fps:29.970
[vo/vdpau/x11] X11 opening display: :0
[vo/vdpau/x11] X11 running at 1280x1024 (":0" => local display)
[vo/vdpau/x11] Detected wm supports NetWM.
[vo/vdpau/x11] Detected wm supports FULLSCREEN state.
[vo/vdpau/x11] Detected wm supports ABOVE state.
[vo/vdpau/x11] Detected wm supports BELOW state.
[vo/vdpau/x11] Display 0 (DVI-I-1): [0, 0, 1280, 1024] @ 60.019740 FPS
[vo/vdpau/x11] Current display FPS: 60.019740
[vo/vdpau] Assuming 60.019740 FPS for framedrop.
[vd] Container reported FPS: 29.970032
[vd] Codec list:
[vd]     lavc:h264 - H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
[vd]     lavc:h264_vdpau (h264) - H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 (VDPAU acceleration)
[vd] Opening video decoder lavc:h264
[vd] Requested hardware decoder not compiled.
[vd] Using software decoding.
[vd] Detected 2 logical cores.
[vd] Requesting 3 threads for decoding.
[vd] Selected video codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 [lavc:h264]
[vo/vdpau/x11] Disabling screensaver.
[ad] Codec list:
[ad]     lavc:flac - FLAC (Free Lossless Audio Codec)
[ad] Opening audio decoder lavc:flac
[ad] Requesting 1 threads for decoding.
[ad] Selected audio codec: FLAC (Free Lossless Audio Codec) [lavc:flac]
[cplayer] Starting playback...
[af] Audio filter chain:
[af]   [in] 48000Hz stereo 2ch s16
[af]   [out] 48000Hz stereo 2ch s16
[af]   [ao] 48000Hz stereo 2ch s16
[ao/pulse] requested format: 48000 Hz, stereo channels, s16
[ao/pulse] Library version: 2.0.0
[ao/pulse] Proto: 26
[ao/pulse] Server proto: 4294967295
[ao/pulse] device buffer: 6000 samples.
[ao/pulse] using soft-buffer of 9600 samples.
[cplayer] AO: [pulse] 48000Hz stereo 2ch s16
[cplayer] AO: Description: PulseAudio audio output
[vf] Video filter chain:
[vf]   [vd] 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[vf]   [in] 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[vf]   [out] 1280x720 yuv420p bt.709/limited CL=mpeg2/4/h264
[cplayer] VO: [vdpau] 1280x720 yuv420p
[cplayer] VO: Description: VDPAU with X11
[vo/vdpau/x11] not waiting for MapNotify
[vo/vdpau] Window size: 1280x720
[vo/vdpau] Video source: 1280x720 (1280x720)
[vo/vdpau] Video display: (0, 0) 1280x720 -> (0, 0) 1280x720
[vo/vdpau] Video scale: 1.000000/1.000000
[vo/vdpau] OSD borders: l=0 t=0 r=0 b=0
[vo/vdpau] Video borders: l=0 t=0 r=0 b=0
[vo/vdpau] Recreating vdpau video mixer.
[global] config path: 'fonts' -> '(NULL)'
[global] config path: 'subfont.ttf' -> '(NULL)'
[global] config path: 'fonts.conf' -> '(NULL)'
[osd/libass] Setting up fonts...
[osd/libass] Done.
[global] config path: 'fonts' -> '(NULL)'
[global] config path: 'subfont.ttf' -> '(NULL)'
[global] config path: 'fonts.conf' -> '(NULL)'
[osd/libass] Setting up fonts...
[osd/libass] Done.
[global] config path: 'fonts' -> '(NULL)'
[global] config path: 'subfont.ttf' -> '(NULL)'
[global] config path: 'fonts.conf' -> '(NULL)'
[osd/libass] Setting up fonts...
[osd/libass] Done.
[statusline] AV: 00:00:28 / 02:56:42 (0%) A-V:  0.000 Cache: 10s+132482KB
[cplayer] Run command: quit, flags=9, args=[0]
[cplayer] EOF code: 6  
[cplayer] 
[ad] Uninit audio filters...
[ad] Uninit audio decoder.
[vd] Uninit video.
[cache] Terminating cache...
[cache] Cache exiting...
[cplayer] 
[cplayer] Exiting... (Quit)
[ytdl_hook] Exiting...
[osc] Exiting...
[ao/pulse] draining...
[vo/vdpau/x11] Enabling screensaver.
[vo/vdpau/x11] uninit ...
```
```
$ mediainfo /media/MMEDIA/DVD/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ BDRip\ 720p\]/Dream\ Theater\ -\ Live\ at\ Budokan\ \(2004\)\ \[2011,\ Progressive\ Metal,\ BDRip\ 720p\].mkv 
General
Unique ID                                : 86456320167382803050727322929349990318 (0x410AE1AFD440631E9D2C2B6B604F17AE)
Complete name                            : /media/MMEDIA/DVD/Dream Theater - Live at Budokan (2004) [2011, BDRip 720p]/Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p].mkv
Format                                   : Matroska
Format version                           : Version 4 / Version 2
File size                                : 6.59 GiB
Duration                                 : 2h 56mn
Overall bit rate mode                    : Variable
Overall bit rate                         : 5 343 Kbps
Movie name                               : Dream Theater - Live at Budokan (2004) [2011, Progressive Metal, BDRip 720p]
Encoded date                             : UTC 2014-03-30 06:18:29
Writing application                      : mkvmerge v6.8.0 ('Theme for Great Cities') 32bit built on Mar  3 2014 15:28:58
Writing library                          : libebml v1.3.0 + libmatroska v1.4.1

Video
ID                                       : 1
Format                                   : AVC
Format/Info                              : Advanced Video Codec
Format profile                           : High@L4.1
Format settings, CABAC                   : Yes
Format settings, ReFrames                : 9 frames
Codec ID                                 : V_MPEG4/ISO/AVC
Duration                                 : 2h 56mn
Nominal bit rate                         : 4 500 Kbps
Width                                    : 1 280 pixels
Height                                   : 720 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Constant
Frame rate                               : 29.970 fps
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Scan type                                : Progressive
Bits/(Pixel*Frame)                       : 0.163
Writing library                          : x264 core 119
Encoding settings                        : cabac=1 / ref=9 / deblock=1:-1:-1 / analyse=0x3:0x113 / me=umh / subme=10 / psy=1 / psy_rd=1,00:0,00 / mixed_ref=1 / me_range=24 / chroma_me=1 / trellis=2 / 8x8dct=1 / cqm=0 / deadzone=21,11 / fast_pskip=0 / chroma_qp_offset=-2 / threads=12 / sliced_threads=0 / nr=0 / decimate=1 / interlaced=0 / bluray_compat=0 / constrained_intra=0 / bframes=8 / b_pyramid=2 / b_adapt=2 / b_bias=0 / direct=3 / weightb=1 / open_gop=0 / weightp=2 / keyint=300 / keyint_min=30 / scenecut=40 / intra_refresh=0 / rc_lookahead=60 / rc=2pass / mbtree=1 / bitrate=4500 / ratetol=1,0 / qcomp=0,60 / qpmin=10 / qpmax=51 / qpstep=4 / cplxblur=20,0 / qblur=0,5 / ip_ratio=1,40 / aq=1:1,00
Language                                 : English
Default                                  : Yes
Forced                                   : No
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709

Audio
ID                                       : 2
Format                                   : FLAC
Format/Info                              : Free Lossless Audio Codec
Codec ID                                 : A_FLAC
Duration                                 : 2h 56mn
Bit rate mode                            : Variable
Channel(s)                               : 2 channels
Sampling rate                            : 48.0 KHz
Bit depth                                : 16 bits
Title                                    : PCM 2.0 1536 Kbps 48.0 KHz 16 bits
Writing library                          : libFLAC 1.3.0 (UTC 2013-05-26)
Language                                 : English
Default                                  : Yes
Forced                                   : No
```

**Comment by wm4** [comment]
It must be a problem with your ffmpeg - it appears it doesn't expose its vdpau API.

mpv build log would be helpful (``build/config.log``).

**Comment by spvkgn** [comment]
Thanks, the issue solved with latest ffmpeg 2.6.2
